File: solutions/go/allergies/1/allergies.go
package allergies

var strAllergies = []string{"eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"}

func Allergies(allergies uint) []string {
	allergicList := make([]string, 0)

	for i := 0; i < len(strAllergies); i++ {
		if allergies&(1<<i) != 0 {
			allergicList = append(allergicList, strAllergies[i])
		}
	}

	return allergicList
}

func AllergicTo(allergies uint, allergen string) bool {
	for i := 0; i < len(strAllergies); i++ {
		if strAllergies[i] == allergen && allergies&(1<<i) != 0 {
			return true
		}
	}

	return false
}

File: solutions/fsharp/allergies/1/Allergies.fs
module Allergies

open System

type Allergen =
    | Eggs = 1
    | Peanuts = 2
    | Shellfish = 4
    | Strawberries = 8
    | Tomatoes = 16
    | Chocolate = 32
    | Pollen = 64
    | Cats = 128

let allergicTo codedAllergies (allergen: Allergen) = codedAllergies &&& int allergen <> 0

let list codedAllergies =
    Enum.GetValues<Allergen>()
    |> Array.filter (allergicTo codedAllergies)
    |> Array.toList
